tasks/image-segmentation-yolov8/src/validate.py

```diff
@@ -0,0 +1,252 @@
+        logging.warning(f">> [Image Segmentation] Failed to upload image \"{plotPath.name}\" with segmentation as artifact")
+    else:
+        logging.info(f">> [Image Segmentation] The segmentation image \"{plotPath.name}\" has been uploaded as artifact")
+
+
+def plotSegmentationImage(
+    imagePath: Path,
+    sampleId: int,
+    origSeg: np.ndarray,
+    predictedMask: np.ndarray,
+    iou: float
+) -> Path:
+
+    img = Image.open(imagePath)
+    fig, axes = plt.subplots(1, 3, figsize = (20, 10))
+
+    axes[0].imshow(img, cmap = "summer")
+    axes[0].set_title(f"Original image, id: {sampleId}")
+    axes[0].axis("off")
+
+    axes[1].imshow(origSeg, cmap = "summer")
+    axes[1].set_title(f"Original segmentation")
+    axes[1].axis("off")
+
+    axes[2].imshow(predictedMask, cmap = "summer")
+    axes[2].set_title(f"Predicted segmentation\nAcc: {iou:.2f}")
+    axes[2].axis("off")
+
+    plotPath = folder_manager.temp.joinpath(f"{sampleId}.jpeg")
+    plt.savefig(plotPath)
+    plt.close()
+    gc.collect()  # The garbage collector is manually invoked after each plotting to prevent a memory leak
+
+    return plotPath
+
+
+def iouScoreClass(testMask: np.ndarray, predictedMask: np.ndarray) -> float:
+    intersecrion = np.logical_and(testMask, predictedMask)
+    union = np.logical_or(testMask, predictedMask)
+
+    return float(np.sum(intersecrion) / np.sum(union) * 100)
+
+
+def maskFromPoly(polygons: list[list[tuple[int, int]]], width: int, height: int) -> np.ndarray:
+    image = Image.new("L", (width, height))
+    for polygon in polygons:
+        draw = ImageDraw.Draw(image)
+        draw.polygon(polygon, fill = 1)
+
+    return np.array(image)
+
+
+def calculateSingleClassAccuracy(
+    classId: int,
+    classIds: list[int],
+    polygons: list[list[tuple[int, int]]],
+    width: int,
+    height: int,
+    origClassSeg: np.ndarray
+) -> float:
+
+    indexes = [i for i, clsId in enumerate(classIds) if clsId == classId]
+    classPolygons = [polygons[i] for i in indexes]
+    predClassMask = maskFromPoly(classPolygons, width, height)
+    iouClass = iouScoreClass(origClassSeg, predClassMask)
+
+    return iouClass
+
+
+def extractPolygonsFromResultsMasks(masks: Masks, height: int, width: int) -> list[list[tuple[int, int]]]:
+    polygons: list[list[tuple[int, int]]] = []
+    for mask in masks:
+        polygon = mask.xyn
+        for points in polygon:
+            poly = points.tolist()
+            poly = [tuple(point) for point in poly]
+            poly = [(int(x * width), int(y * height)) for (x, y) in poly]
+            polygons.append(poly)
+
+    return polygons
+
+
+def processSampleResult(
+    taskRun: TaskRun[ImageDataset],
+    result: Results,
+    sample: ImageSample
+) -> dict[str, str]:
+
+    sampleAnnotation = sample.load().annotation
+    if sampleAnnotation is None:
+        raise RuntimeError(f">> [Image Segmentation] The sample named \"{sample.name}\" (id: \"{sample.id}\") has no annotation.")
```

Comment:
If a sample has no annotation that means its groundtruth mask is zeroes (background)